### PR TITLE
feat(addon-doc): add `TUI_DOC_EXCLUDED_PROPERTIES` token to exclude inherited props

### LINKED ISSUE #3241: 🚀 - allow exclusions from common documentation (TextfieldControllerDocumentation)
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

addon-doc

### Description

add input [exclusion] for common documentation (TextfieldControllerDocumentation)

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3241

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/addon-doc/src/tokens/excluded-properties.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {InjectionToken, Provider} from '@angular/core';
+
+export const TUI_DOC_EXCLUDED_PROPERTIES = new InjectionToken<readonly string[]>(
+    `[TUI_DOC_EXCLUDED_PROPERTIES]: [TUI_PROPERTY_EXCLUSION]: Token to exclude inherited documentation properties`,
```
Comment (splincode) [inline projects/addon-doc/src/tokens/excluded-properties.ts:4]: `TUI_DOC_EXCLUDED_PROPERTIES`

File: projects/addon-doc/src/tokens/excluded-properties.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    {factory: () => []},
+);
+
+export function tuiExcludeProperties(useValue: readonly string[]): Provider {
```
Comment (splincode) [inline projects/addon-doc/src/tokens/excluded-properties.ts:8]: `tuiDocExcludeProperties`

File: projects/addon-doc/src/components/documentation/documentation.component.ts [hunk 3]
```diff
@@ -59,6 +61,8 @@ export class TuiDocDocumentationComponent implements AfterContentInit {
         @Inject(ChangeDetectorRef) private readonly changeDetectorRef: ChangeDetectorRef,
         @Inject(TUI_DOC_DOCUMENTATION_TEXTS)
         readonly texts: [string, string, string, string, string],
+        @Inject(TUI_DOC_EXCLUDED_PROPERTIES)
+        readonly excludedProperties: readonly string[],
```
Comment (splincode) [inline projects/addon-doc/src/components/documentation/documentation.component.ts:65]: why array instead of set?

**Review by splincode — APPROVED** [review]

File: projects/addon-doc/src/components/documentation/documentation.component.ts [hunk 4]
```diff
@@ -82,6 +86,13 @@ export class TuiDocDocumentationComponent implements AfterContentInit {
         return this.isAPI ? this.texts[0] : this.texts[1];
     }
 
+    matcher: TuiMatcher<TuiDocDocumentationPropertyConnectorDirective<any>> = (
+        item: TuiDocDocumentationPropertyConnectorDirective<any>,
+        exclusions: readonly string[],
```
Comment (splincode) [inline projects/addon-doc/src/components/documentation/documentation.component.ts:91]: `exclusions: Set<string>,`

File: projects/addon-doc/src/components/documentation/documentation.component.ts [hunk 5]
```diff
@@ -82,6 +86,13 @@ export class TuiDocDocumentationComponent implements AfterContentInit {
...
+    ) => {
+        return !exclusions.includes(item.documentationPropertyName);
```
Comment (splincode) [inline projects/addon-doc/src/components/documentation/documentation.component.ts:93]: `return !exclusions.has(item.documentationPropertyName)`

File: projects/addon-doc/src/tokens/excluded-properties.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import {InjectionToken, Provider} from '@angular/core';
+
+export const TUI_DOC_EXCLUDED_PROPERTIES = new InjectionToken<readonly string[]>(
+    `[TUI_DOC_EXCLUDED_PROPERTIES]: [TUI_PROPERTY_EXCLUSION]: Token to exclude inherited documentation properties`,
+    {factory: () => []},
```
Comment (splincode) [inline projects/addon-doc/src/tokens/excluded-properties.ts:5]: `{factory: () => new Set()},`

File: projects/addon-doc/src/tokens/excluded-properties.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+);
+
+export function tuiExcludeProperties(useValue: readonly string[]): Provider {
+    return {provide: TUI_DOC_EXCLUDED_PROPERTIES, useValue};
```
Comment (splincode) [inline projects/addon-doc/src/tokens/excluded-properties.ts:9]: `export function tuiExcludeProperties(values: readonly string[]): Provider {`

`return {provide: TUI_DOC_EXCLUDED_PROPERTIES, useValue: new Set(values) };`

File: projects/addon-doc/src/components/documentation/documentation.component.ts (hunk 3 above)
Comment (vladimirpotekhin) [reply projects/addon-doc/src/components/documentation/documentation.component.ts]: as we discussed before, `set` has no performance advantage over `array` https://jsbench.me/6zlbp925nn/1
but ok

File: projects/addon-doc/src/components/documentation/documentation.component.ts [hunk 8]
```diff
@@ -82,6 +86,13 @@ export class TuiDocDocumentationComponent implements AfterContentInit {
         return this.isAPI ? this.texts[0] : this.texts[1];
     }
 
+    matcher: TuiMatcher<TuiDocDocumentationPropertyConnectorDirective<any>> = (
+        item: TuiDocDocumentationPropertyConnectorDirective<any>,
+        exclusions: Set<string>,
+    ) => {
+        return !exclusions.has(item.documentationPropertyName);
```
Comment (waterplea) [inline projects/addon-doc/src/components/documentation/documentation.component.ts:93]: Unnecessary `return`

**Review by waterplea — APPROVED** [review]

**Pull request merged**